src/Routing/Route/Route.php

```diff
@@ -705,11 +709,13 @@ public function match(array $url, array $context = []): ?string
             return null;
         }
 
-        // check patterns for routed params
+        // check patterns for routed params, if they are not the default value
         if (!empty($this->options)) {
             foreach ($this->options as $key => $pattern) {
-                if (isset($url[$key]) && !preg_match('#^' . $pattern . '$#u', (string)$url[$key])) {
-                    return null;
+                if (!array_key_exists($key, $defaults) || $defaults[$key] !== $url[$key]) {
+                    if (isset($url[$key]) && !preg_match('#^' . $pattern . '$#u', (string)$url[$key])) {
```

Comment:
Could these conditions be combined with those on the line above?

Reply:
Done.
I had them separate because I thought it will be easier to follow the change and it would be easier to read.